genesis/engine/entities/rigid_entity/rigid_link.py

```diff
@@ -121,82 +122,88 @@ def _build(self):
         for vgeom in self._vgeoms:
             vgeom._build()
 
-        self._init_mesh = self._compose_init_mesh()
-
-        # inertial_mass and inertia_i
-        if self._inertial_mass is None:
-            if len(self._geoms) == 0 and len(self._vgeoms) == 0:
-                self._inertial_mass = 0.0
+        if self._inertial_mass is None or self._inertial_pos is None or self._inertial_i is None:
+            # Determine which geom list to use: geoms first, then vgeoms, then fallback
+            if len(self._geoms) > 0:
+                geom_list = self._geoms
+                get_mesh = lambda geom: trimesh.Trimesh(geom.init_verts, geom.init_faces)
+                get_pos = lambda geom: geom._init_pos
+                get_quat = lambda geom: geom._init_quat
+            elif len(self._vgeoms) > 0:
```

Comment:
It is not clear if collision geometry of visual geometry should be used to compute such property. I have the feeling that collision geometries may be very off. There are often rough approximation using primitives, unless they are the output of convex decomposition. You can easily check in metadata of meshes whether they are the output of convex decomposition and prefer them in such a case (or more generally if they are true meshes instead of primitive geometries), otherwise ressort on visual. Another option would be to add some flag in morph to control whether collision or visual should be preferred in such a case, but I would rather avoid that.

I would recommend adding some info message to explain that spatial inertia informations are being deduced from the geometry.